lib/problems.rb

```diff
@@ -1,9 +1,41 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) because it  goes through each item
+# Space Complexity: could be up to o(n) because im building a stack.
 def balanced(string)
```

Comment:
👍   One suggestion, you could have a hash where the keys are the closing braces and the values are the open braces.  That could help you dry up your if-else statements

```ruby
brace_matches = {
  ")" => "(",
  "]" => "[",
  "}" => "{",
}
```